### PR TITLE
Added DAG class and Node class

### LINKED ISSUE #9: Create Node and DAG classes for clustering purposes

### PR DESCRIPTION
**Pull request by rpjayasekara**
This PR resolves #9 

- Node class includes all the attributes to represent a Node in a workflow object form.

- DAG class includes all the attributes and methods to represent a workflow DAG in object form.

Script path -
"swarmform/core/swarm_dag.py"

### CODE REVIEW

File: swarmform/core/swarm_dag.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+class DAG:
+
+    def __init__(self, sf):
+
+        """
+        Args:
+            sf(SwarmFlow): SwarmFlow object
+        """
+
+        self._dag_id = sf.wf_id
```
Comment (KalanaDananjaya) [inline swarmform/core/swarm_dag.py]: it will be sf.wf_id

File: swarmform/core/swarm_dag.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        self._dag_name = sf.name or "Unnamed-WF"
```
Comment (KalanaDananjaya) [inline swarmform/core/swarm_dag.py]: Shall we rename as Unnamed SF to follow firework's naming format and to maintain a difference in names of a workflow and a swarmflow?

**Review by KalanaDananjaya — CHANGES_REQUESTED** [review]

File: swarmform/core/swarm_dag.py (hunk 1 above)
Comment (rpjayasekara) [reply swarmform/core/swarm_dag.py]: sf_id right?

File: swarmform/core/swarm_dag.py (hunk 2 above)
Comment (rpjayasekara) [reply swarmform/core/swarm_dag.py]: noted

File: swarmform/core/swarm_dag.py (hunk 1 above)
Comment (AyeshW) [reply swarmform/core/swarm_dag.py]: Yes, sf_id

**Review by AyeshW — COMMENTED** [review]
Ok for me, Please do the @KalanaDananjaya 's requested changes

**Review by KalanaDananjaya — APPROVED** [review]

**Review by AyeshW — APPROVED** [review]

**Pull request merged**